### PR TITLE
lock.py: allow selection of machines to list by os-type/os-version

### PR DESCRIPTION
**Pull request by dmick**
Also, clean up the status selection a tiny bit with winnow()

Signed-off-by: Dan Mick <dan.mick@redhat.com>

### CODE REVIEW

**Comment by liewegas** [comment]
works for me (to list only rhel 7.1 machines)!

File: teuthology/lock.py [hunk 1]
```diff
@@ -151,6 +151,18 @@ def json_matching_statuses(json_file_or_str, statuses):
     return return_statuses
 
 
+def winnow(statuses, arg, status_key, func=None):
```
Comment (zmc) [inline teuthology/lock.py]: A docstring would be really nice here!

Comment (dmick) [inline teuthology/lock.py]: yeah...

Comment (dmick) [inline teuthology/lock.py]: forcepushed

Comment (zmc) [inline teuthology/lock.py]: thanks!

**Pull request merged**